code/game/machinery/dehumidifier.dm

```diff
@@ -0,0 +1,115 @@
+//This magic dehumidifier is a portable water-to-nothing machine, although it does need a charged cell to work.
+//See drain.dm!
+/obj/machinery/dehumidifier
+	use_power = POWER_USE_OFF
+	anchored = 0
+	density = 1
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "dhum-off"
+	name = "emergency dehumidifier"
+	desc = "A portable dehumidifier with emergency high-volume pump to help keep you dry when it's getting damp... just very slowly."
+	var/obj/item/weapon/cell/cell = /obj/item/weapon/cell/high
+	var/on = 0
+	var/drainage = 1
+	var/last_gurgle = 0
+	atom_flags = ATOM_FLAG_CLIMBABLE
+	clicksound = "switch"
+
+/obj/machinery/dehumidifier/Initialize()
+	if(ispath(cell))
+		cell = new cell(src)
+	. = ..()
+	update_icon()
+
+/obj/machinery/dehumidifier/on_update_icon(var/rebuild_overlay = 0)
+	if(!on)
+		icon_state = "dhum-off"
+	else
+		icon_state = "dhum-on"
+
+	if(rebuild_overlay)
+		cut_overlays()
+		if(panel_open)
+			add_overlay("dhum-open")
+
+/obj/machinery/dehumidifier/examine(mob/user)
+	. = ..(user)
```

Comment:
Need to check if the returned value if true before displaying the rest of the examination.